src/Command/ListTasksCommand.php

```diff
@@ -102,20 +105,39 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         }
 
         $symfonyStyle->success(sprintf('%d task%s found', count($tasks), count($tasks) > 1 ? 's' : ''));
-
         $table = new Table($output);
-        $table->setHeaders(['Name', 'Description', 'Expression', 'Last execution date', 'Next execution date', 'Last execution duration', 'Last execution memory usage', 'State', 'Tags']);
-        $table->addRows(array_map(fn (TaskInterface $task): array => [
-            $task->getName(),
-            $task->getDescription() ?? 'No description set',
-            $task->getExpression(),
-            null !== $task->getLastExecution() ? $task->getLastExecution()->format(DATE_ATOM) : 'Not executed',
-            (new CronExpression($task->getExpression()))->getNextRunDate()->format(DATE_ATOM),
-            null !== $task->getExecutionComputationTime() ? Helper::formatTime($task->getExecutionComputationTime() / 1000) : 'Not tracked',
-            null !== $task->getExecutionMemoryUsage() ? Helper::formatMemory($task->getExecutionMemoryUsage()) : 'Not tracked',
-            $task->getState(),
-            implode(', ', $task->getTags()) ?? 'No tags set',
-        ], $tasks));
+        $table->setHeaders(['Type', 'Name', 'Description', 'Expression',  'Last execution date', 'Next execution date', 'Last execution duration', 'Last execution memory usage', 'State', 'Tags']);
+
+        array_walk($tasks, function (TaskInterface $task) use ($table): void {
+            $table->addRow([
+                (new ReflectionClass($task))->getShortName(),
+                $task->getName(),
+                $task->getDescription() ?? 'No description set',
+                $task->getExpression(),
+                null !== $task->getLastExecution() ? $task->getLastExecution()->format(DATE_ATOM) : 'Not executed',
+                (new CronExpression($task->getExpression()))->getNextRunDate()->format(DATE_ATOM),
+                null !== $task->getExecutionComputationTime() ? Helper::formatTime($task->getExecutionComputationTime() / 1000) : 'Not tracked',
+                null !== $task->getExecutionMemoryUsage() ? Helper::formatMemory($task->getExecutionMemoryUsage()) : 'Not tracked',
+                $task->getState(),
+                implode(', ', $task->getTags()) ?? 'No tags set',
+            ]);
+
+            if ($task instanceof ChainedTask) {
+                $table->addRows(array_map(fn (TaskInterface $task): array => [
```

Comment:
I wonder if it could be a good idea to add a method `map()` inside `TaskList`, this way, we can iterate directly via the list without importing the method and/or leaving the list, any thought? 🤔

Reply:
yes I think it's a good idea, in any case we already have a walk method so why not a map method

Reply:
Hello @Guikingone 👋 

After some thought, I wonder if implementing the `map ()` method is really a good idea because it won't be able to return a `TaskListInterface` but a simple array.
What will be the added value of this method ?

WDYT ?

Reply:
Hi 👋🏻 

IMHO, that's not an issue to worry about, the `TaskListInterface` can either return a list, an array or even an integer, that not something that lead to an issue.

The value is mainly the developer experience as we often use `array_map` on the list in the bundle and personally, I prefer to use `list->map(...)` rather than `array_map($list->toArray()...)`, in the end, the behaviour still pretty must the same but the first syntax is "better" to read 🤔

Reply:
it's ok for me, i will work on it